identity-iota/src/did/url/iota_did.rs

```diff
@@ -248,7 +248,8 @@ impl IotaDID {
 
   /// Returns the Tangle `network` of the `DID`.
   pub fn network(&self) -> Network {
-    Network::from_name(self.network_str())
+    // SAFETY: `network_str` never returns an empty string.
+    Network::from_name(self.network_str()).unwrap()
   }
```

Comment:
This should probably return a ` Result` too, not all non-empty network strings will be valid networks (according to our spec).